### PR TITLE
Add labels section on pod level for  operator and pilot components

### PR DESCRIPTION
**Pull request by vedmichv**
Add labels on pod level for istio-operator and istiod. The additional labels might be required to configure logging and monitoring systems (fluentd as an example)

[ ] Configuration Infrastructure
[ ] Docs
[ X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request.

[X] Does not have any user-facing changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: releasenotes/notes/33733.yaml
Comment (shamsher31) [inline releasenotes/notes/33733.yaml:8]: Either link or issue number is sufficient.
```suggestion
```

**Comment by vedmichv** [comment]
@shamsher31  could you please review my PR?

**Comment by jacob-delgado** [comment]
@vedmichv I've put this on the agenda for the environments WG this week to get resolution for this.

**Comment by vedmichv** [comment]
@jacob-delgado  thanks, waiting for any feedback.

**Comment by costinm** [comment]
As discussed in WG - great idea, but please use 'standard' naming, like 'labels' top level

**Comment by costinm** [comment]
and maybe annotations too ?

**Comment by vedmichv** [comment]
> and maybe annotations too ?

Hi @costinm , I added annotations for istio-operator chart. On istio-control has been already implemented.

**Comment by jacob-delgado** [comment]
> > and maybe annotations too ?
> 
> Hi @costinm , I added annotations for istio-operator chart. On istio-control has been already implemented.

The feedback from the WG is that this is a nice addition, but we would like some consistency with charts as they are made using `helm create <chart>`.

File: manifests/charts/istio-control/istio-discovery/templates/deployment.yaml
Comment (jacob-delgado) [inline manifests/charts/istio-control/istio-discovery/templates/deployment.yaml:47]: rename `pilotLabels` to `labels` or `selectorLabels`.mSame for `operatorLabels`

Comment (vedmichv) [reply manifests/charts/istio-control/istio-discovery/templates/deployment.yaml]: @jacob-delgado  I renamed it. Could you please review it one more time.

**Review by shamsher31 — APPROVED** [review]

File: manifests/charts/istio-operator/templates/deployment.yaml
Comment (shamsher31) [reply manifests/charts/istio-operator/templates/deployment.yaml:21]: Just to avoid empty annotations like [this](https://github.com/istio/istio/pull/34003/files#diff-76349a7ea6a7900cdb39f2f768922855162b2865dfb591f31b78ef5f26405b27R171).

File: manifests/charts/istio-operator/values.yaml
Comment (howardjohn) [inline manifests/charts/istio-operator/values.yaml:29]: This doesn't not follow standard helm practices. There is no need to prefix with `operator` - its already in the operator chart, its redundant.

File: operator/pkg/apis/istio/v1alpha1/values_types.proto
Comment (howardjohn) [inline operator/pkg/apis/istio/v1alpha1/values_types.proto:644]: IMO this is dangerous and not a good practice. Selector labels are immutable. 

We can add pod labels without making them selector labels.

File: manifests/charts/istio-operator/values.yaml
Comment (howardjohn) [reply manifests/charts/istio-operator/values.yaml:29]: We should be consistent with the `selectorLabels` (although I have concerns with that as well).

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: operator/pkg/apis/istio/v1alpha1/values_types.proto
Comment (vedmichv) [reply operator/pkg/apis/istio/v1alpha1/values_types.proto:644]: @howardjohn could you please share your thoughts, what'll be the way to add labels for pilot.  The improvement adds labels for pod which allows it on gitops tools like ArgoCD. In my case, I tried to make the changes for the pods after provisioning, but argoCD rollback.

File: manifests/charts/istio-operator/values.yaml
Comment (vedmichv) [reply manifests/charts/istio-operator/values.yaml]: Renamed

File: operator/pkg/apis/istio/v1alpha1/values_types.proto
Comment (pdrastil) [reply operator/pkg/apis/istio/v1alpha1/values_types.proto:644]: @howardjohn You are completely right - I'm team member of @vedmichv and we found out yesterday by issue on prod that [Istio gateway spec](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#GatewaySpec) `labels` are actually selector + pod labels which is not clearly documented. When we've tried to add a new label Istio operator was not able to replace / patch manifest and ended in error state.

**Review by howardjohn — APPROVED** [review]

**Comment by shamsher31** [comment]
ping @istio/wg-docs-maintainers @istio/wg-policies-and-telemetry-maintainers

**Review by Xunzhuo — APPROVED** [review]

**Review by jacob-delgado — APPROVED** [review]

**Comment by jacob-delgado** [comment]
cc @douglas-reid @mandarjog @nrjpoddar Can we get a review for this from the observability maintainers?

**Pull request merged**
